### PR TITLE
Adding NPD

### PR DESCRIPTION
**Pull request by Sanjay1611**
Signed-off-by: Sanjay Nathani <sanjay.nathani@mayadata.io>

This PR adds NPD as an optional deployment in the chart.
The other custom configurations can also be edited and added in the `values.yaml` accordingly from [here](https://artifacthub.io/packages/helm/deliveryhero/node-problem-detector)

### CODE REVIEW

**Comment by avishnu** [comment]
Do we need to upgrade the chart version? There is a linting error..

File: deploy/charts/openebs-monitoring/Chart.yaml [hunk 1]
```diff
@@ -46,3 +46,7 @@ dependencies:
     version: "16.5.*"
     repository: https://prometheus-community.github.io/helm-charts
     condition: kube-prometheus-stack.install
+  - name: node-problem-detector
+    version: 2.0.2
```
Comment (rajaSahil) [inline deploy/charts/openebs-monitoring/Chart.yaml]: Use wildcards for specifying version (`version: "2.0.*" `)
Helm chart best practices: [versions](https://helm.sh/docs/chart_best_practices/dependencies/#versions)

**Comment by rajaSahil** [comment]
@Sanjay1611 Please add all the configuration used in values.yaml in the README.

File: deploy/charts/openebs-monitoring/values.yaml [hunk 2]
```diff
@@ -160,6 +160,23 @@ kube-prometheus-stack:
...
+  extraVolumes:
+  - name: kmsg
+    hostPath:
+      path: /dev/kmsg
+
+  extraVolumeMounts:
+    - name: kmsg
+      mountPath: /dev/kmsg
+      readOnly: true
+
```
Comment (Ab-hishek) [inline deploy/charts/openebs-monitoring/values.yaml]: Can you also provide comments here regarding what these fields are used for?

**Comment by Ab-hishek** [comment]
@Sanjay1611 You can also update all the metrics provided by npd in the docs folder as a sepearte md file.

**Comment by Sanjay1611** [comment]
> @Sanjay1611 You can also update all the metrics provided by npd in the docs folder as a sepearte md file.

@Ab-hishek I think instead of maintaining a sepearte md file for these metrics we can directly attach the link to parameters for the user to visit. Like for npd we can attach this link https://artifacthub.io/packages/helm/deliveryhero/node-problem-detector
Helps a user to have a wider understanding.

**Comment by Ab-hishek** [comment]
This [link]( https://artifacthub.io/packages/helm/deliveryhero/node-problem-detector) provides only helm configuration fields. Metrics docs would contain all the metrics(or events) which node problem detector exports which will be used for monitoring.

File: deploy/charts/openebs-monitoring/values.yaml [hunk 3]
```diff
@@ -173,6 +173,25 @@ kube-prometheus-stack:
       - grafana-polystat-panel
       - snuids-trafficlights-panel
 
+## Configuration for node-problem-detector subchart
+node-problem-detector:
```
Comment (rajaSahil) [inline deploy/charts/openebs-monitoring/values.yaml:177]: Can you also add these configurations to the configuration table?

**Review by Ab-hishek — APPROVED** [review]

**Review by rajaSahil — APPROVED** [review]

**Comment by avishnu** [comment]
The pending review comments will be addressed in subsequent PRs.

**Pull request merged**